lib/web_of_science/links_client.rb

```diff
@@ -0,0 +1,85 @@
+module WebOfScience
+  # Links AMR (Article Match Retrieval) Service
+  # @see http://ipscience-help.thomsonreuters.com/LAMRService/WebServicesOverviewGroup/overview.html Service documentation
+  class LinksClient
+
+    LINKS_HOST = 'https://ws.isiknowledge.com'.freeze
+    LINKS_PATH = '/cps/xrpc'.freeze
+
+    attr_reader :username, :password, :host
+
+    # @param [String] username
+    # @param [String] password
+    # @param [String] host
+    def initialize(username: nil, password: nil, host: LINKS_HOST)
+      @host = host
+      @username = username
+      @password = password
+      (@username, @password) = Base64.decode64(Settings.WOS.AUTH_CODE).split(':', 2) unless username
+    end
+
+    # Retrieve identifier 'fields' for the record 'ids'
+    # @param [Array<String>] ids
+    # @param [Array<String>] fields (defaults to ['ut', 'doi', 'pmid'])
+    # @return [Hash<String => Hash>]
+    def links(ids, fields = %w(ut doi pmid))
+      ids.uniq!
+      raise ArgumentError, '1-50 ids required' if ids.empty? || ids.count > 50
+      raise ArgumentError, 'fields cannot be empty' if fields.empty?
```

Comment:
Consider using `#blank?` instead of `#empty?` to guard against `nil`s here?

Reply:
OK